bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/internal/CBusHandlerFactory.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.cbus.internal;
+
+import java.util.Hashtable;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.cbus.CBusBindingConstants;
+import org.openhab.binding.cbus.handler.CBusCGateHandler;
+import org.openhab.binding.cbus.handler.CBusDaliHandler;
+import org.openhab.binding.cbus.handler.CBusLightHandler;
+import org.openhab.binding.cbus.handler.CBusNetworkHandler;
+import org.openhab.binding.cbus.handler.CBusTemperatureHandler;
+import org.openhab.binding.cbus.handler.CBusTriggerHandler;
+import org.openhab.binding.cbus.internal.discovery.CBusGroupDiscovery;
+import org.openhab.binding.cbus.internal.discovery.CBusNetworkDiscovery;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link CBusHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Scott Linton - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.cbus")
+public class CBusHandlerFactory extends BaseThingHandlerFactory {
+
+    private @Nullable ServiceRegistration<?> cbusCGateHandlerServiceReg = null;
+    private @Nullable ServiceRegistration<?> cbusNetworkHandlerServiceReg = null;
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return CBusBindingConstants.SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+        if (thingTypeUID.equals(CBusBindingConstants.BRIDGE_TYPE_CGATE)) {
```

Comment:
Generically I would say:

Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

In this case `thingTypeUID` is not Nullable but I would still it looks better but, not essential.

Reply:
All changed